samples/bluetooth/hci_uart/src/main.c

```diff
@@ -324,6 +328,64 @@ void bt_ctlr_assert_handle(char *file, uint32_t line)
 }
 #endif /* CONFIG_BT_CTLR_ASSERT_HANDLER */
 
+#if defined(CONFIG_VS_HCI_SYSTEM_FATAL_ERROR_HANDLER)
+void k_sys_fatal_error_handler(unsigned int reason, const z_arch_esf_t *esf)
+{
+	/* 0xAA + r0/a1 + 0xAA + r1/a2 + 0xAA + r2/a3 + 0xAA + r3/a4 + 0xAA + r12/ip +
+	 * 0xAA + r14/lp + 0xAA + XPSR
+	 */
+	uint32_t len = 7 * (1 + 4);
+
+	/* Disable interrupts, this is unrecoverable */
+	(void)irq_lock();
+
+	uart_irq_rx_disable(hci_uart_dev);
+	uart_irq_tx_disable(hci_uart_dev);
+
+	if (esf != NULL) {
+		uart_poll_out(hci_uart_dev, H4_EVT);
+		/* Vendor-Specific debug event */
+		uart_poll_out(hci_uart_dev, 0xff);
+
+		uart_poll_out(hci_uart_dev, len);
+		uart_poll_out(hci_uart_dev, 0xAA);
+
+		net_buf_add_u8(buf, 0xAA);
+		uart_poll_out(hci_uart_dev, esf->basic.a1);
+
+		net_buf_add_u8(buf, 0xAA);
+		uart_poll_out(hci_uart_dev, esf->basic.a2);
+
+		net_buf_add_u8(buf, 0xAA);
+		uart_poll_out(hci_uart_dev, esf->basic.a3);
+
+		net_buf_add_u8(buf, 0xAA);
+		uart_poll_out(hci_uart_dev, esf->basic.a4);
+
+		net_buf_add_u8(buf, 0xAA);
+		uart_poll_out(hci_uart_dev, esf->basic.ip);
+
+		net_buf_add_u8(buf, 0xAA);
+		uart_poll_out(hci_uart_dev, esf->basic.lr);
+
+		net_buf_add_u8(buf, 0xAA);
+		uart_poll_out(hci_uart_dev, esf->basic.xpsr);
+
+		/* Send the event over rpmsg */
+		hci_rpmsg_send(buf);
+	}
+
+	LOG_PANIC();
+	LOG_ERR("Halting system");
+
+	for (;;) {
+		/* Spin endlessly */
+	}
```

Comment:
IIRC `while (true)` is preferred over `for (;;)`.

Reply:
Do you know why, ie is it a technical difference or a zephyr style thing ? cos I say the for(;;) a lot in embedded code.

Reply:
Technically the will both do the same. 

I think it's more of a readability "issue", and that `for` loops are typically used to "do `x`, `y` number of times", so using it for infinite loops is somewhat semantically odd :) 

I can't find any coding guidelines to back up my claim, so I clearly did not remember correctly :D

Reply:
Yeah I remember being a bit dumbfounded as why the writers used `for` instead of `while` the first time I came across one of those. I think you're right, I don't remember seeing this `for(;;)` construct in zephyr.

Reply:
> Yeah I remember being a bit dumbfounded as why the writers used `for` instead of `while` the first time I came across one of those. I think you're right, I don't remember seeing this `for(;;)` construct in zephyr.

A quick search yields 38 in Zephyr compare to nearly 600 of `while (true/1)`, so the latter is certainly more used :)